recommend/runtimePolicy.go

```diff
@@ -61,40 +63,65 @@ func createRuntimePolicy(img *ImageInfo) error {
 			Type:      "file",
 		})
 		if err != nil {
-			return err
+			continue
 		}
 		sumResp = append(sumResp, resp)
 	}
 
-	ms, err := checkProcessFileData(sumResp)
+	ms, err := checkProcessFileData(sumResp, img.Distro)
 	if err != nil {
 		return err
 	}
-	img.writePolicyFile(ms)
+
+	img.writePolicyFile(*ms)
 	return nil
 }
 
-func checkProcessFileData(sumResp []*opb.Response) (MatchSpec, error) {
-	var filePaths SysRule
-	ms := MatchSpec{}
-	ms.OnEvent.Severity = 1
-	ms.Description.Tldr = "Kubernetes serviceaccount folder access should be limited "
-	ms.Name = "audit-serviceaccount-runtime"
-	ms.OnEvent.Tags = []string{"KUBERNETES", "SERVICE ACCOUNT", "RUNTIME POLICY"}
-	ms.OnEvent.Message = "serviceaccount access detected"
-	ms.OnEvent.Action = "Audit"
+func checkProcessFileData(sumResp []*opb.Response, distro string) (*MatchSpec, error) {
+	var filePaths types.KnoxSys
+	fromSourceArr := []types.KnoxFromSource{}
+	ms := MatchSpec{
+		Name: "audit-serviceaccount-runtime",
+		Description: Description{
+			Tldr: "Kubernetes serviceaccount folder access should be limited",
+		},
+	}
 	for _, eachResp := range sumResp {
 		for _, fileData := range eachResp.FileData {
 			if strings.HasPrefix(fileData.ProcName, saPath[0]) || strings.HasPrefix(fileData.ProcName, saPath[1]) {
-				filePaths.FromSource = append(filePaths.FromSource, fileData.ParentProcName)
+				if strings.HasSuffix(fileData.ParentProcName, "/") {
+					fromSourceArr = append(fromSourceArr, types.KnoxFromSource{
+						Dir: fileData.ParentProcName,
+					})
+				} else {
+					fromSourceArr = append(fromSourceArr, types.KnoxFromSource{
+						Path: fileData.ParentProcName,
+					})
+				}
 			}
 		}
 	}
-	filePaths.Path = saPath
-	filePaths.Recursive = true
+	if len(fromSourceArr) < 1 {
+		log.Info("No serviceaccount access detected. Skipping runtime policy creation on ", distro)
+		return &MatchSpec{}, nil
```

Comment:
Creating an empty `MatchSpec` object and returning the pointer of that object doesn't make any sense. You should use `nil` here and handle the nil-case in the parent function.